File: src/c6/SearchEngineForm.java
package c6;

import java.io.IOException;
import java.io.PrintWriter;

import javax.servlet.ServletException;
import javax.servlet.http.*;

import util.SearchSpec;
import util.SearchUtilities;
import util.ServletUtilities;

public class SearchEngineForm extends HttpServlet {
	public void doGet(HttpServletRequest req, HttpServletResponse res) 
			throws ServletException, IOException {
		res.setContentType("text/html");
		PrintWriter out = res.getWriter();
		String title = "One-Stop Web Search!";
		String actionURL = "/csj/c6/searchengines";
		out.println(ServletUtilities.headWithTitle(title) +
				"<body bgcolor=\"#FDF5E6\">\n" +
				"<center>\n" +
				"<h1>" + title + "</h1>\n" +
				"<form action=\"" + actionURL + "\">\n" +
				"  Search keywords: \n" +
				"  <input type=\"text\" name=\"searchString\"><p>\n");
		SearchSpec[] specs = SearchUtilities.getCommonSpecs();
		for (int i = 0; i < specs.length; i++) {
			String searchEngineName = specs[i].getName();
			out.println("<input type=\"radio\" " + 
					"name=\"searchEngine\" " +
					"value=\"" + searchEngineName + "\">\n");
			out.println(searchEngineName + "<br>\n");
		}
		out.println("<br>  <input type=\"submit\">\n" +
				"</form>\n" +
				"</center></body></html>");
	}
}

File: src/gocode/c6/search_engine_form.go
package c6

import (
	"fmt"
	"net/http"

	"../util"
)

type SearchEngineForm struct{}

func (sef *SearchEngineForm) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	var err error
	switch r.Method {
	case "GET":
		err = sef.handleGet(w, r)
	case "POST":
		err = sef.handlePost(w, r)
	default:
		err = fmt.Errorf("%v", r.Method+" may not be supported")
	}
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}

func (sef *SearchEngineForm) handleGet(w http.ResponseWriter,
	r *http.Request) (err error) {
	w.Header().Set("Content-Type", "text/html")
	title := "One-Stop Web Search!"
	actionURL := "/csj/c6/searchengines"
	w.Write([]byte(util.HeadWithTitle(title) +
		"<body bgcolor=\"#FDF5E6\">\n" +
		"<center>\n" +
		"<h1>" + title + "</h1>\n" +
		"<form action=\"" + actionURL + "\">\n" +
		"  Search keywords: \n" +
		"  <input type=\"text\" name=\"searchString\"><p>\n\n"))
	specs := util.GetCommonSpecs()
	for i := 0; i < len(specs); i++ {
		w.Write([]byte("<input type=\"radio\" name=\"searchEngine\" " +
			"value=\"" + specs[i].Name + "\">\n\n"))
		w.Write([]byte(specs[i].Name + "<br>\n\n"))
	}
	w.Write([]byte("<br>  <input type=\"submit\">\n" +
		"</form>\n" +
		"</center></body></html>\n\n"))

	return
}

func (sef *SearchEngineForm) handlePost(w http.ResponseWriter,
	r *http.Request) (err error) {
	err = sef.handleGet(w, r)
	return
}
